core/capabilities/ccip/ccipsolana/executecodec.go

```diff
@@ -0,0 +1,258 @@
+	}
+
+	messages := []cciptypes.Message{
+		{
+			Header: cciptypes.RampMessageHeader{
+				MessageID:           executeReport.Message.Header.MessageId,
+				SourceChainSelector: cciptypes.ChainSelector(executeReport.Message.Header.SourceChainSelector),
+				DestChainSelector:   cciptypes.ChainSelector(executeReport.Message.Header.DestChainSelector),
+				SequenceNumber:      cciptypes.SeqNum(executeReport.Message.Header.SequenceNumber),
+				Nonce:               executeReport.Message.Header.Nonce,
+				MsgHash:             cciptypes.Bytes32{},        // todo: info not available, but not required atm
+				OnRamp:              cciptypes.UnknownAddress{}, // todo: info not available, but not required atm
+			},
+			Sender:         executeReport.Message.Sender,
+			Data:           executeReport.Message.Data,
+			Receiver:       executeReport.Message.LogicReceiver.Bytes(),
+			ExtraArgs:      buf.Bytes(),
+			FeeToken:       cciptypes.UnknownAddress{}, // <-- todo: info not available, but not required atm
+			FeeTokenAmount: cciptypes.BigInt{},         // <-- todo: info not available, but not required atm
+			TokenAmounts:   tokenAmounts,
+		},
+	}
+
+	offchainTokenData := make([][][]byte, 0, 1)
+	if executeReport.OffchainTokenData != nil {
+		offchainTokenData = append(offchainTokenData, executeReport.OffchainTokenData)
+	}
+
+	proofs := make([]cciptypes.Bytes32, 0, len(executeReport.Proofs))
+	for _, proof := range executeReport.Proofs {
+		proofs = append(proofs, proof)
+	}
+
+	chainReport := cciptypes.ExecutePluginReportSingleChain{
+		SourceChainSelector: cciptypes.ChainSelector(executeReport.SourceChainSelector),
+		Messages:            messages,
+		OffchainTokenData:   offchainTokenData,
+		Proofs:              proofs,
+	}
+
+	report := cciptypes.ExecutePluginReport{
+		ChainReports: []cciptypes.ExecutePluginReportSingleChain{chainReport},
+	}
+
+	return report, nil
+}
+
+func parseExtraArgsMap(input map[string]any) (ccip_router.SVMExtraArgs, error) {
+	// Parse input map into SolanaExtraArgs
+	var out ccip_router.SVMExtraArgs
+
+	// Iterate through the expected fields in the struct
+	for fieldName, fieldValue := range input {
+		switch fieldName {
+		case "ComputeUnits":
+			// Expect uint32
+			if v, ok := fieldValue.(uint32); ok {
+				out.ComputeUnits = v
+			} else {
+				return out, errors.New("invalid type for ComputeUnits, expected uint32")
+			}
+		case "IsWritableBitmap":
+			// Expect uint64
+			if v, ok := fieldValue.(uint64); ok {
+				out.IsWritableBitmap = v
+			} else {
+				return out, errors.New("invalid type for IsWritableBitmap, expected uint64")
+			}
+		case "Accounts":
+			// Expect [][32]byte
+			if v, ok := fieldValue.([][32]byte); ok {
+				accounts := make([]solana.PublicKey, len(v))
+				for i, val := range v {
+					accounts[i] = solana.PublicKeyFromBytes(val[:])
+				}
+				out.Accounts = accounts
+			} else {
+				return out, errors.New("invalid type for Accounts, expected [][32]byte")
+			}
+		}
```

Comment:
Add a default case too, and return error from there.
We shouldn't ever see any unknown field here.